### PR TITLE
feat: Tabs included in the filter for a dashboard.

### PR DESCRIPTION
**Pull request by amitkharb**
Closes: https://github.com/lightdash/lightdash/issues/9659

### Description:
If the dashboard contains tabs, then the filter will show the tiles of each tab and user can create a filter for specific tab. 
If there are no tabs, then there are no changes in the existing functionality

Screenshot 2024-04-24 at 18 30 59

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by hamzahc1** [comment]
Once we can build a render environment @PriPatel - would you want to test the functionality?

**Comment by rephus** [comment]
@amitkharb some tests are failing, it looks like you didn't run the pre-commit hook to format the files. Let me know if you need any help fixing those.

**Comment by rephus** [comment]
This PR https://github.com/lightdash/lightdash/pull/9878 is already introducing the activeTabUuid (using localstorage) , I'm not sure if this is going to create conflicts

**Comment by amitkharb** [comment]
> @amitkharb some tests are failing, it looks like you didn't run the pre-commit hook to format the files. Let me know if you need any help fixing those.

Let me have a look, I do not know how it missed the pre-commit hook, it ran for the previous PR. Thanks

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx [hunk 1]
```diff
@@ -36,6 +40,8 @@ type Props = {
 
 const TileFilterConfiguration: FC<Props> = ({
     tiles,
+    tabs,
+    activeTabUuid,
```
Comment (rephus) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx:44]: are these 2 props used ?

Comment (amitkharb) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/TileFilterConfiguration.tsx:44]: yes, 

tabs are used to create the Accordion panels and activeTabUUid is to open the Accordion panel if tabuuid === activeTabUuid

**Comment by amitkharb** [comment]
> > @amitkharb some tests are failing, it looks like you didn't run the pre-commit hook to format the files. Let me know if you need any help fixing those.
> 
> Let me have a look, I do not know how it missed the pre-commit hook, it ran for the previous PR. Thanks

Did another commit to this PR, and looks like Pre Commit Hook has ran 
Screenshot 2024-04-25 at 15 40 12

**Review by PriPatel — COMMENTED** [review]
Hey @amitkharb Appreciate this very much!  I have one styling change and one UX request:

### Styling change:
- [ ] Change the tab Names in the accordions to `14px` and `500` font-weight.

---
### Ux change
I have one UX change because in the original ticket, you should have been able to select/deselect tabs in one go. After playing around with this, I think we can make this even easier for the user.
Screenshot 2024-04-25 at 16 12 00
- [ ] Add a toggle next to each accordion. 
- [ ] **If the toggle is off,** the filter does not apply to the whole tab and any future tiles added to the tab should not be affected. Disable the accordion
- [ ] **If the toggle is on**, there should be no change to the current behaviour
- [ ] Add relevant tooltips as shown in the screenshot.

Please feel free to Slack me if you want a better explanation of the above! Thank you so much

**Comment by rephus** [comment]
> Did another commit to this PR, and looks like Pre Commit Hook has ran ![Screenshot 2024-04-25 at 15 40 12](https://private-

mm, I'm not sure what's happening, are you running the commit from the root folder ? if you us ea subfolder, it might skip the pre-hook. Do you have husky installed ? 

Either way, could you run `yarn workspace backend format -w`  and commit those changes ?

**Comment by amitkharb** [comment]
> > Did another commit to this PR, and looks like Pre Commit Hook has ran ![Screenshot 2024-04-25 at 15 40 12](https://private-
> 
> mm, I'm not sure what's happening, are you running the commit from the root folder ? if you us ea subfolder, it might skip the pre-hook. Do you have husky installed ?
> 
> Either way, could you run `yarn workspace backend format -w` and commit those changes ?

I am making few more changes as requested by Priyanka and then will do another commit

**Comment by rephus** [comment]
> I am making few more changes as requested by Priyanka and then will do another commit

Hi @amitkharb are you still working on this ? let me know if you need help or another review.

**Comment by amitkharb** [comment]
I have done all the changes requested above. 
Only thing I can not get the Tooltip working for the Switch button but then found this on Internet https://github.com/mantinedev/mantine/issues/4757
We are using mantine version 6.0.21 and issue was fixed in version 7

**Comment by rephus** [comment]
I tested this on render, something doesn't work as expected with these switch buttons. They act like `radio buttons` (only 1 can be disabled at the same time) 

See video: 

[Screencast from 01-05-24 17:36:50.webm](https://github.com/lightdash/lightdash/assets/1983672/2c29aef5-3100-43cb-8be0-358b90448595)

**Review by rephus — CHANGES_REQUESTED** [review]
See my previous comment, I think the UI is not working as expected. 

Suggestion: revert to the previous version without the `switch`

**Comment by rephus** [comment]
Yes, I tested again and it is now working as expected . Good job

**Pull request merged**